### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by hugelgupf**
Sorry for the PR spam. Apparently I don't know how to use GitHub.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Gopkg.toml example
```
Comment (rminnich) [inline Gopkg.toml]: this is not an example any more, is it?

File: options.go [hunk 2]
```diff
@@ -72,34 +66,47 @@ func (o Options) Get(key OptionCode) ([]byte, error) {
...
 		}
 
+		if !buf.Has(length) {
+			return io.ErrUnexpectedEOF
+		}
+
 		// N bytes: option data
-		data := buf.Consume(int(length))
+		data := buf.Consume(length)
 		if data == nil {
```
Comment (rminnich) [inline options.go]: not unexpectedEOF?

File: packet.go [hunk 3]
```diff
@@ -9,7 +9,15 @@ import (
 
 const (
 	minPacketLen = 236
-	MAClen       = 16
+	macLen       = 16
```
Comment (rminnich) [inline packet.go]: where's this number come from? It doesn't match any mac len I'm used to? The RFC?

Comment (rminnich) [reply packet.go]: is it one mac address, two mac addresses, ...

File: packet_test.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+			packet: func() dhcp4.Packet {
+				p := dhcp4.NewPacket(dhcp4.BootRequest)
+				p.SetBroadcast(true)
+				return p
+			},
+			want: &Packet{
+				Op:        BootRequest,
+				HType:     1,
+				Broadcast: true,
+				CIAddr:    net.IP{0, 0, 0, 0},
```
Comment (rminnich) [inline packet_test.go]: I think you might want to give these addresses some non-zero values ....

Comment (rminnich) [reply packet_test.go]: I realize that's a pain in the neck.

File: options.go (hunk 2 above)
Comment (hugelgupf) [reply options.go]: Actually the Has check above should catch this anyway, but fixed.

File: Gopkg.toml (hunk 1 above)
Comment (hugelgupf) [reply Gopkg.toml]: dep automatically adds this stuff. removed.

**Comment by hugelgupf** [comment]
ptal?

**Comment by hugelgupf** [comment]
I'm just gonna chump it in, I need it now.

**Pull request merged**